docs/common/craft-parts/reference/plugins/poetry_plugin.rst

```diff
@@ -0,0 +1,105 @@
+.. _craft_parts_poetry_plugin:
+
+Poetry plugin
+=============
+
+**Version**: 2.1
+
+The Poetry plugin can be used for Python projects that use the `Poetry`_ build system.
+
+Keywords
+--------
+
+This plugin uses the common :ref:`plugin <part-properties-plugin>` keywords as
+well as those for :ref:`sources <part-properties-sources>`.
+
+Additionally, this plugin provides the plugin-specific keywords defined in the
+following sections.
+
+poetry-with:
+~~~~~~~~~~~~
+**Type:** list of strings
+
+Extra dependency groups to use other than the defaults.
```

Comment:
On line 30 we provide Poetry's docs about env vars, but here we assume the user is familiar with dependency groups. What do you think about linking to https://python-poetry.org/docs/managing-dependencies#dependency-groups here?

Reply:
I don't see a reason not to link it here.